### PR TITLE
Pods 3939

### PR DESCRIPTION
**Pull request by vinnicombe**

### CODE REVIEW

File: test/controllers/actions/MutableFakeDataRetrievalAction.scala [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class MutableFakeDataRetrievalAction extends DataRetrievalAction {
+  def setDataToReturn(userAnswers: Option[UserAnswers]): Unit = dataToReturn = userAnswers
+
+  private var dataToReturn: Option[UserAnswers] = None
+
+  override protected def transform[A](request: IdentifierRequest[A]): Future[OptionalDataRequest[A]] =
+    dataToReturn match {
```
Comment (grantrobson) [inline test/controllers/actions/MutableFakeDataRetrievalAction.scala]: Don't need to bother doing a match here. Just return optional data request with whatever is in dataToReturn.

File: test/controllers/chargeD/DeleteMemberControllerSpec.scala [hunk 2]
```diff
@@ -42,24 +44,30 @@ import uk.gov.hmrc.viewmodels.{NunjucksSupport, Radios}
 import scala.concurrent.Future
 
 class DeleteMemberControllerSpec extends ControllerSpecBase with MockitoSugar with NunjucksSupport with JsonMatchers with OptionValues with TryValues {
+  private val mutableFakeDataRetrievalAction: MutableFakeDataRetrievalAction = new MutableFakeDataRetrievalAction()
   private val mockAftConnector: AFTConnector = mock[AFTConnector]
+  private val application: Application = applicationBuilderMutableRetrievalAction(mutableFakeDataRetrievalAction)
+    .overrides(
+      bind[AFTConnector].toInstance(mockAftConnector)
```
Comment (grantrobson) [inline test/controllers/chargeD/DeleteMemberControllerSpec.scala]: You could just pass the extra binding into the extraModules parameter of the data retrieval action, i.e. applicationBuilderMutableRetrievalAction(mutableFakeDataRetrievalAction, Seq(bind[AFTConnector].toInstance(mockAftConnector)))

File: test/controllers/actions/MutableFakeDataRetrievalAction.scala (hunk 1 above)
Comment (vinnicombe) [reply test/controllers/actions/MutableFakeDataRetrievalAction.scala]: Done

File: test/controllers/chargeD/DeleteMemberControllerSpec.scala (hunk 2 above)
Comment (vinnicombe) [reply test/controllers/chargeD/DeleteMemberControllerSpec.scala]: Done

**Pull request merged**